Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -260,23 +260,32 @@ In order to address these attack vectors, the outcome of the following checks sh
 
 #### Overview
 
--- TODO [Provide a general description of the issue "Testing WebView Protocol Handlers".] --
+Protocol handler is a basic form of IPC in iOS system. Basically, it allows for invoking arbitrary applications, by calling a URL scheme with specified parameters. There are some default handlers<sup>[1]</sup>, however a developer is allowed to register his own custom URL scheme<sup>[2]</sup>. Unfortunately, this brings a serious security concern, what found out developers of Skype application<sup>[3]</sup>. The Skype application registered `skype://` protocol handler, which allows for making a call to arbitrary number without asking for a user's permission. Attackers exploited this vulnerability by putting an invisible `<iframe src=”skype://xxx?call"></iframe>` (where `xxx` was replaced by a premium number), so any Skype's user who visited malicious website unconsciously was forced to call to a premium number.
 
 #### Static Analysis
 
--- TODO [Describe how to assess this given either the source code or installer package (APK/IPA/etc.), but without running the app. Tailor this to the general situation (e.g., in some situations, having the decompiled classes is just as good as having the original source, in others it might make a bigger difference). If required, include a subsection about how to test with or without the original sources.] --
+A security concern related with protocol handlers, should be arised, when the URL is not validated or the user is not prompted for confirmation in the application before making a particular action. 
```

Comment:
You're right there's a mistake. What do you think about such sentence: "A security concern related with protocol handlers should arise, when...blablabla"?